contessa/runner.py

```diff
@@ -25,22 +31,30 @@ def run(
         self,
         raw_rules: List[Dict[str, str]],
         check_table: Dict,
-        result_table: Dict,  # todo - docs for quality name, maybe defaults..
+        result_table: Optional[
+            Dict
+        ] = None,  # todo - docs for quality name, maybe defaults..
         context: Optional[Dict] = None,
     ):
         check_table = Table(**check_table)
-        result_table = ResultTable(**result_table, model_cls=self.model_cls)
         context = self.get_context(check_table, context)
 
         normalized_rules = self.normalize_rules(raw_rules)
         refresh_executors(check_table, self.conn, context)
-        quality_check_class = self.get_quality_check_class(result_table)
-        self.conn.ensure_table(quality_check_class.__table__)
+
+        if result_table:
+            result_table = ResultTable(**result_table, model_cls=self.model_cls)
+            quality_check_class = self.get_quality_check_class(result_table)
+            self.conn.ensure_table(quality_check_class.__table__)
+        else:
+            quality_check_class = CheckResult
```

Comment:
you're requiring the same interface, but nothing really says there should be. I understand we are using python, but I shouldn't we unify the models for the quality check class?